src/main/java/pages/JobPage.java

```diff
@@ -0,0 +1,88 @@
+package pages;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.testng.Assert;
+
+import java.util.List;
+import java.util.Random;
+
+public class JobPage extends BasePage{
+
+    private String jobCheckStr = "//*[text()='%s']/following-sibling::span";
+    private final By JobsLoc = By.xpath("//*[@class='job_list_company_title']");
+    private String checkLocStr = "//*[@id='jobsfilter-category']//*[text()='%s']/preceding-sibling::input";
+    private final WebDriverWait wait;
+    private final By randomJobLoc = By.xpath("//*[@class='job-inner job-item-title']");
+    private final By randomJobTitleLoc = By.xpath(".//*[@class='font_bold']");
+    private String jobTitleFirst;
+    private WebElement jobRandom;
```

Comment:
You shouldn't declare WebElement variable here. It's an error prone, when you put a reference on an element in DOM and don't take into account that page can be updated during further actions. In that case, your code will throw StaleReferenceElement exception.